backend/api/Controllers/MissionController.cs

```diff
@@ -141,20 +216,114 @@ public async Task<ActionResult<byte[]>> GetMap([FromRoute] string assetCode, str
     }
 
     /// <summary>
-    /// Schedule a new mission
+    /// Reschedule an existing mission
     /// </summary>
     /// <remarks>
-    /// <para> This query schedules a new mission and adds it to the database </para>
+    /// <para> This query reschedules an existing mission and adds it to the database </para>
+    /// </remarks>
+    [HttpPost("{missionId}/reschedule")]
+    [Authorize(Roles = Role.User)]
+    [ProducesResponseType(typeof(MissionRun), StatusCodes.Status201Created)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<MissionRun>> Rechedule(
+        [FromRoute] string missionId,
+        [FromBody] RescheduleMissionQuery rescheduledMissionQuery
+    )
+    {
+        var robot = await _robotService.ReadById(rescheduledMissionQuery.RobotId);
+        if (robot is null)
+            return NotFound($"Could not find robot with id {rescheduledMissionQuery.RobotId}");
+
+        MissionDefinition? missionDefinition;
+        try
+        {
+            missionDefinition = await _missionDefinitionService.ReadById(missionId);
+            if (missionDefinition == null)
+                return NotFound("Mission definition not found");
+        }
+        catch (HttpRequestException e)
```

Comment:
Why are we catching and `HttpRequestException` here? The underlying functionality here is only calling the database for a `MissionDefinition`